src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -3436,5 +3455,736 @@ public void Deconstruct(out int a, out int b)
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var firstX = tree.GetRoot().DescendantNodes().OfType<SingleVariableDesignationSyntax>().Where(d => d.Identifier.ValueText == "x").First();
+            var firstXSymbol = model.GetDeclaredSymbol(firstX);
+            Assert.Equal("System.Int32 Script.x", firstXSymbol.ToTestDisplayString());
+            Assert.Equal(SymbolKind.Field, firstXSymbol.Kind);
+
+            var secondX = tree.GetRoot().DescendantNodes().OfType<SingleVariableDesignationSyntax>().Where(d => d.Identifier.ValueText == "x").Skip(1).First();
+            var secondXSymbol = model.GetDeclaredSymbol(secondX);
+            Assert.Equal("System.String Script.x", secondXSymbol.ToTestDisplayString());
+            Assert.Equal(SymbolKind.Field, secondXSymbol.Kind);
+
+            Assert.NotEqual(firstXSymbol, secondXSymbol);
+        }
+
+        [Fact]
+        public void UnassignedUsedInDeconstructionInScript()
+        {
+            var source =
+@"
+System.Console.Write(x);
+var (x, y) = (1, 2);
+";
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "0");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var x = GetDeconstructionVariable(tree, "x");
+            var xSymbol = model.GetDeclaredSymbol(x);
+            var xRef = GetReference(tree, "x");
+            Assert.Equal("System.Int32 Script.x", xSymbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x, xRef);
+        }
+
+        [Fact]
+        public void FailedInferenceInDeconstructionInScript()
+        {
+            var source =
+@"
+var (x, y) = (1, null);
+";
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.GetDeclarationDiagnostics().Verify();
+
+            comp.VerifyDiagnostics(
+                // (2,1): error CS8130: The type information on the left-hand-side 'y' and right-hand-side 'null' of the deconstruction was insufficient to infer a merged type.
+                // var (x, y) = (1, null);
+                Diagnostic(ErrorCode.ERR_DeconstructCouldNotInferMergedType, "var (x, y) = (1, null);").WithArguments("y", "null").WithLocation(2, 1)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var x = GetDeconstructionVariable(tree, "x");
+            var xSymbol = model.GetDeclaredSymbol(x);
+            Assert.Equal("var Script.x", xSymbol.ToTestDisplayString());
+            var xType = ((FieldSymbol)xSymbol).Type;
+            Assert.True(xType.IsErrorType());
```

Comment:
Let's open an issue to track that.

---
In reply to: [80325077](https://github.com/dotnet/roslyn/pull/13957#discussion_r80325077) [](ancestors = 80325077)